### ISSUE TITLE
Zero change to bower_components/*

### ISSUE THREAD

**Issue by jhs** [jeffmjack/template-bootstrap#1]
Wow! Yeah this is nice! Thirty-nine commits already. That's awesome.

You're free to do what you want of course, but I really think the basic policy should be: "never change anything in bower_components/*".

It seems like all work should be sibling to bower_components, and then that stuff should call into bower_components, right?

So like for example, `/index.html`

~~~ html
<script src="bower_components/jquery/jquery.js"></script>
<style src="bower_components/bootstrap/whatever.css</style>
~~~

Am I crazy? I know you're symlinking and doing all this stuff. Symlinks don't even exist on Windows. So that can't be what the mainstream world is doing. What specific problems happen to you if you adopt this policy?

**Comment by jeffmjack** [comment]
I agree with you about keeping things out of bower_components, although that's a little bit tricky.

So, I just moved `variables.less` and similar project-specific .less files out of `bower_components` and into `/public/less-editables`, and just made the `bootstrap.less` file (in bower_components) import those files from the `public/less-editables/` subdir.

The problem is that `bootstrap.less` itself would be overridden in the event of a `bower install` and would start failing silently - the new `bootstrap/less` directory would have the boilerplate `bootstrap.less` file, importing all the boilerplate `variables.less` files as if everything were correct.

It's nice that the original `variables.less` file and its cohorts would still exist in `public/less-editables`, but the app itself would still be broken in the event of a `bower install`, and because we haven't moved `bootstrap.less` out of `bower_components`, we're still not treating `bower_components` like a read-only directory, like we should be.

**Comment by jeffmjack** [comment]
One solution, which I am resistant to, would be to setup grunt (or gulp, at this point) to exist in `/public` and just have expectations about how a freshly-installed `bower_components/bootstrap` would look and behave accordingly.

That said, this all seems like a lot of work for the off-chance that someone erroneously runs `bower install bootstrap` and is unwilling to use git to rollback lost files.

**Comment by jhs** [comment]
Maybe it would help to back up a bit and say, what are your real goals.

Because, to me, like, the point of this project is to do the hard and tedious work once, and have it done for all time. So, why not just have your Gruntfile reference a bunch of stuff in bower_components/foo/blah and all of that compiles into your top-level code, for example `/final.css` which is the sum total of all the bootstrap stuff, plus your own LESS files?

Again, I know neither Grunt nor Bootstrap, but, it can only really have CSS, JavaScript, and HTML, and of course you will be changing the HTML. So why not just have Grunt pull the CSS and JS from bootstrap and be done?
